### PR TITLE
himalaya: fix broken config to support 0.6.x

### PR DESCRIPTION
**Pull request by toastal**
### Description

Himalaya had breaking config changes from 0.5.x to 0.6.x. @ambroisie mentioned wanting to step down from maintaining the module.

This isn't backwards compatible because the new config isn't. It now explicitly requires `backend` and `sender` are set to an enum which requires certain properties. I suppose `backend = "imap"` and `sender = "smtp"` could be `default` values to follow what was previously set as the only and required options, but I believe it would be better to be explicit moving forwards--just like `himalaya`.

This fix allowed me to verify that the Vim plugin still worked too https://github.com/NixOS/nixpkgs/pull/195927/files

### Checklist

- [ ] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [ ] Code tested through `nix-shell --pure tests -A run.all`. (failed with `neovim`)
- [x] Code tested through `nix-shell --pure tests -A run.himalaya`.

- [x] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/docs/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [x] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

  - [x] Added myself and the module files to `.github/CODEOWNERS`.

### Comments

Please offer a non-proprietary option for contribution instead of being limited to the Microsoft-owned GitHub forge. Identity is tied to a Microsoft GitHub account too which is as barrier to contribution and requires giving away ones data.

I don't like the use of `nixfmt` because it makes changes that break `git` diffs. I prefer `nixpkgs-fmt` as it has a philosophy of "expand, never contract" and so lines never get pulled up and cuddled if the string or set is now shorter making it harder to review.

### CODE REVIEW

File: modules/programs/himalaya.nix
Comment (ambroisie) [inline modules/programs/himalaya.nix:53]: You only check the settings existence but don't add it to the configuration?

Comment (toastal) [reply modules/programs/himalaya.nix:53]: @ambroisie  I'm not entirely sure how to do this in Nix syntax... it's like an ADT where some setting are dependent on both the `sender` and `backend` properties.

```
type Sender =
  | IMAP { imap-port :: Number, ...etc }
  | Sendmail { sendmail-cmd :: String }
```

Comment (toastal) [reply modules/programs/himalaya.nix:53]: In this case, the `sendmail-cmd` is not needed or used if the `sender` is `imap`.

Comment (ambroisie) [reply modules/programs/himalaya.nix:53]: Yes I understood that part, my confusion was just that `sendmail-cmd` seemed to not be set, but that's only because I forgot that `// account.himalaya.settings` is done at the end.

Comment (ambroisie) [reply modules/programs/himalaya.nix:53]: Now that I am looking at this again, I'm thinking that perhaps either the `throw` should be removed (and let `himalaya` itself complain about the missing command) or perhaps it should be an actual assertion, using the module system.

Will let the maintainers weight in about what's usually done in home-manager.

Comment (toastal) [reply modules/programs/himalaya.nix:53]: I'm okay with any of those options.

Comment (toastal) [reply modules/programs/himalaya.nix]: I did the latter... removed the `-cmd` check

**Review by ambroisie — APPROVED** [review]

**Comment by rycee** [comment]
Thanks! Is this all good for merge?

About contributions outside GitHub you can try sending patches to my public inbox on sr.ht: https://lists.sr.ht/~rycee/public-inbox For now that is an unofficial channel, though.

**Comment by toastal** [comment]
@rycee I've been using it for a while now. There seems to be a bug inside the Vim plugin for me, but that's not related to getting the config up and running so I could launch `himalaya` to see even see that error.

**Comment by NomisIV** [comment]
I hate to be pushy, but I'm waiting for this to be merged as I'm using himalaya as my main email client. Are there any blockers for merging?

**Comment by toastal** [comment]
I've been trying to do the occasional rebase so folks could use in the interim. It's worked fine for me for couple of months now.
-- 
toastal ไข่ดาว | https://toast.al
PGP: 7944 74b7 d236 dab9 c9ef  e7f9 5cce 6f14 66d4 7c9e

**Comment by teto** [comment]
@rycee do we follow some convention to signal breaking changes ? in "conventionalcommits", you can detect breaking changes by looking for BREAKING CHANGE in the last line of the commit. The first line contains also a `!`. Maybe we could adopt some scheme like that ? that's keeping me from merging. We need some kind of log for breaking changes between releases like nixos does

**Comment by rycee** [comment]
Thanks! Merged to master now.

**Pull request merged**
